types/signed_header_test.go

```diff
@@ -0,0 +1,104 @@
+package types
+
+import (
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestVerify(t *testing.T) {
+	trusted, privKey, err := GetRandomSignedHeader()
+	require.NoError(t, err)
+	time.Sleep(time.Second)
+	untrustedAdj, err := GetNextRandomHeader(trusted, privKey)
+	require.NoError(t, err)
+	tests := []struct {
+		prepare func() *SignedHeader
+		err     bool
```

Comment:
[Follow Up]

we should have the expected error here. Then in the test execution we are validating that we are erroring exactly as expected and are not erroring for some other reason. This will help better future proof the code if there are functions that have a chain of checks and we re-order them.

Reply:
Created issue here: https://github.com/rollkit/rollkit/issues/1049